platform/posix/transport/src/plaintext_posix.c

```diff
@@ -224,77 +185,34 @@ int32_t Plaintext_Send( NetworkContext_t * pNetworkContext,
-    sendTimeoutLen = ( socklen_t ) sizeof( sendTimeout );
-    getTimeoutStatus = getsockopt( pPlaintextParams->socketDescriptor,
-                                   SOL_SOCKET,
-                                   SO_SNDTIMEO,
-                                   &sendTimeout,
-                                   &sendTimeoutLen );
-
-    /* Make #select return immediately if getting the timeout failed. */
-    if( getTimeoutStatus < 0 )
-    {
-        sendTimeout.tv_sec = 0;
-        sendTimeout.tv_usec = 0;
-    }
 
-    /* MISRA Directive 4.6 flags the following line for a violation of using a
-     * basic type "int" rather than a type that includes size and signedness information.
-     * We suppress the violation as the flagged type, "fd_set", is a POSIX
-     * system-specific type, and is used for the call to "select()". */
-
-    /* MISRA Rule 14.4 flags the following line for using condition expression "0"
-     * as a boolean type. We suppress the violation as the "FD_ZERO" is a POSIX
-     * specific macro utility whose implementation is supplied by the system.
-     * The "FD_ZERO" macro is called as specified by the POSIX manual here:
-     * https://pubs.opengroup.org/onlinepubs/009695399/basedefs/sys/select.h.html */
-    /* coverity[misra_c_2012_directive_4_6_violation] */
-    /* coverity[misra_c_2012_rule_14_4_violation] */
-    FD_ZERO( &writefds );
-
-    /* MISRA Directive 4.6 flags the following line for a violation of using a
-     * basic type "int" rather than a type that includes size and signedness information.
-     * We suppress the violation as the flagged type, "fd_set", is a POSIX
-     * system-specific type, and is used for the call to "select()". */
-
-    /* MISRA Rule 10.1, Rule 10.8 and Rule 13.4 flag the following line for
-     * implementation of the "FD_SET()" POSIX macro. We suppress these violations
-     * as "FD_SET" is a POSIX specific macro utility whose implementation
-     * is supplied by the system.
-     * The "FD_ZERO" macro is used as specified by the POSIX manual here:
-     * https://pubs.opengroup.org/onlinepubs/009695399/basedefs/sys/select.h.html */
-    /* coverity[misra_c_2012_directive_4_6_violation] */
-    /* coverity[misra_c_2012_rule_10_1_violation] */
-    /* coverity[misra_c_2012_rule_13_4_violation] */
-    /* coverity[misra_c_2012_rule_10_8_violation] */
-    FD_SET( pPlaintextParams->socketDescriptor, &writefds );
-    /* Check if data can be written to the socket. */
-    selectStatus = select( pPlaintextParams->socketDescriptor + 1,
-                           NULL,
-                           &writefds,
-                           NULL,
-                           &sendTimeout );
-
-    if( selectStatus > 0 )
+    /* Initialize the file descriptor. */
+    pollFds.events = POLLOUT;
+    pollFds.revents = 0;
+    /* Set the file descriptor for poll. */
+    pollFds.fd = pPlaintextParams->socketDescriptor;
+
+    /* Check if there is data to read from the socket. */
```

Comment:
```suggestion
    /* Check if socket is available to send data on the network. This is done to avoid blocking on send() call when the socket is ready to accept more data to send on the network (possibly due to a full TX buffer) . */
```